examples/vision/sample_size_estimate.py

```diff
@@ -0,0 +1,601 @@
+    val_ds = (
+        val_ds.map(
+            lambda img, lab: (tf.image.resize(img, image_size), lab),
+            num_parallel_calls=AUTO,
+        )
+        .map(
+            lambda img, lab: (img, tf.one_hot(lab, num_classes)),
+            num_parallel_calls=AUTO,
+        )
+        .cache()
+        .batch(batch_size)
+        .prefetch(AUTO)
+    )
+    return train_ds, val_ds, nr_of_subsamples
+
+
+def train_model(train_ds, val_ds, test_ds):
+    """
+    1) Builds a VGG19 model.
+    2) Trains only the top layers for 10 epochs.
+    3) Unfreezes from block 4 onwards.
+    4) Trains the deeper layers for 20 more epochs.
+    """
+
+    model = build_model("VGG19", num_classes)
+
+    # Compile and train top layers
+    history = compile_and_train(
+        model,
+        train_ds,
+        val_ds,
+        metrics=[keras.metrics.AUC(name="auc"), "acc"],
+        optimizer=keras.optimizers.Adam(),
+        patience=3,
+        epochs=10,
+    )
+
+    # Unfreeze model from block 4 onwards
+    model = unfreeze(model, "block4")
+
+    # Compile and train for 20 epochs with a lower learning rate
+    fine_tune_epochs = 20
+    total_epochs = history.epoch[-1] + fine_tune_epochs
+
+    history_fine = compile_and_train(
+        model,
+        train_ds,
+        val_ds,
+        metrics=[keras.metrics.AUC(name="auc"), "acc"],
+        optimizer=keras.optimizers.Adam(lr=1e-4),
+        patience=5,
+        epochs=total_epochs,
+    )
+
+    # Calculate model accuracy on the test set
+    y_pred = model.predict(test_ds)
+    y_pred_indices = np.argmax(y_pred, axis=1)
+    acc = np.sum(np.equal(y_pred_indices, y_true_indices)) / num_test_samples
+    return acc
+
+
+"""
+Let's create a small subset and plot a few images from the training set. These should now
+be augmented using RandAug.
+"""
+
+# Test image augmentations on a subset (5% of training data)
+train_ds, val_ds, nr_of_samples = create_subsets(train_data, 0.1)
+
+# Plot images from the training set
+image_batch, label_batch = next(iter(train_ds))
+
+plt.figure(figsize=(16, 12))
+for n in range(30):
+    ax = plt.subplot(5, 6, n + 1)
+    plt.imshow(image_batch[n].numpy().astype("int32"))
+    plt.title(np.array(class_names)[label_batch[n].numpy() == True][0])
+    plt.axis("off")
+
+"""
+# Train models iteratively
+
+Now that we have model building functions and supporting iterative functions we can train
+the model over several subsample splits.
+
+- We select the subsample splits as 5%, 10%, 25% and 50% of the downloaded dataset. We
+pretend that only 50% of the actual data is available at present.
+- We train the model 5 times from scratch at each split and record the accuracy values.
+
+Note that this trains 20 models and will take some time. Make sure you have a GPU runtime
+active.
+
+To keep this example lightweight, sample data from a previous training run is provided
+after the commented out code below.
+"""
+
+# # Train all the sample models and calculate accuracy
```

Comment:
Rather than commenting an entire code block, put it in a function then point out to readers that the data you hard-code is supposed to be the output of the function.